### ISSUE TITLE
Assistant `/file` command should hold reference rather than copy text into buffer

### ISSUE THREAD

**Issue by yaroslavyaroslav** [zed-industries/zed#16742]
### Description

Currently when user runs `/file` command zed copies the content of a given files to the chat history, and this behaviour, if user uses inline assistant or workflow to modify those files (which is the most use case for those agentish features so far) brings those particular files content of those has been imported right away in outdated state, which leads to flawed feedback loop from AI model perspective as it don't see actual code base to work with.

So currently user has to actually reapply that command manually every time on the file changes.

### If applicable, add mockups / screenshots to help present your vision of the feature

So the idea here is to make /file command to hold reference to the current state of the file till it's presented on the disc, and to update it's content implicitly in AI pane after each actual update of the file.

**Comment by knyazer** [comment]
I suggest to actually have a different command for this, e.g. ```attach_file```: if using the messaging API for the AI models, having frequent context changes might lead to an increase in costs. 

This happens because most of the providers utilize KV caching under the hood, and if you change the context (e.g. restart the chat) you invalidate the cache.

**Comment by yaroslavyaroslav** [comment]
Fair point, but I still see a little sense to keep cached invalid yet cheap content within the chat history to keep it as default behaviour.

 If user is using the chat without relying on agentish features cache on a service side shouldn't be invalidated as files hasn't been changed. On the other hand using chat with that agentish feature breaks content-model feedback loop as the latter don't see actual state of the file update to what it is suggesting. Thus cached outdated content even a cheep one will decrease robustness of the model provided solution.

**Comment by notpeter** [comment]
While this is potentially an interesting feature, it's not something we intend to prioritize in the near future.  The new agent panel can update the model as files have have changed.

Thanks for reporting.
